pkg/provisioner/ironic/ironic.go

```diff
@@ -456,6 +456,25 @@ func (p *ironicProvisioner) ValidateManagementAccess(data provisioner.Management
 		// below.
 	}
 
+	// If no PreprovisioningImage builder is enabled we set the Node network_data
+	// this enables Ironic to inject the network_data into the ramdisk image
+	if !p.config.havePreprovImgBuilder {
+		networkDataRaw := data.PreprovisioningNetworkData
+		if networkDataRaw != "" {
+			var networkData map[string]interface{}
+			if yamlErr := yaml.Unmarshal([]byte(networkDataRaw), &networkData); yamlErr != nil {
+				p.log.Info("failed to unmarshal networkData from PreprovisioningNetworkData")
+				result, err = transientError(errors.Wrap(err, "invalid preprovisioningNetworkData"))
```

Comment:
nit: use the standard `fmt.Errorf` instead of `errors.Wrap`

Reply:
Thanks, copy-paste mistake, I also forgot to change `err` to `yamlErr` - now fixed!